src/main/java/com/sourcegraph/webhook/WebhookListener.java

```diff
@@ -35,8 +38,30 @@
     }
 
     @EventListener
-    public void onPullRequestEvent(PullRequestActivityEvent event) {
-        String key = "pr:activity:status";
+    public void onPullRequestEvent(PullRequestEvent event) {
+        String key = "pr:event";
+
+        // When review status changes, two events are fired in most cases
+        // 1. A PR Activity Event
+        // 2. A PR Participant Event
+        // We only want to forward the Activity events.
+        // However, when moving from Needs Works to Unapproved, no activity event is fired
+        // and so in this one case we want to forward the Participant Event
+        if (event instanceof PullRequestParticipantStatusUpdatedEvent) {
+            PullRequestAction action = event.getAction();
+            PullRequestParticipantStatus previousStatus = ((PullRequestParticipantStatusUpdatedEvent) event).getPreviousStatus();
+            if (action != PullRequestAction.UNAPPROVED) {
+                return;
+            }
+            if (previousStatus != PullRequestParticipantStatus.NEEDS_WORK) {
+                return;
+            }
+            key =  "pr:participant:status";
```

Comment:
Do we need a new key? Or can't we reuse the `pr:activity:reviewers` key that I see further down? I'd rather have use introduce some abstraction here, than to handle this special case here and in our codebase.

Reply:
Our current Go code mostly ignores those keys so I'll take this opportunity to clean it up. We don't actually care what kind of activity it is. 

We do still need the new key though so that we know when we've received a PullRequestParticipantStatusUpdatedEvent

Reply:
Ah, we use those strings here too so I'll leave them there for now:
https://github.com/sourcegraph/bitbucket-server-plugin/blob/a312d669568da1aedf4903b9855ff2198b588bd2/src/main/java/com/sourcegraph/webhook/EventSerializer.java#L74

I wouldn't mind doing a proper refactor of this code but at this point I don't see it as a priority.